src/bootstrap/min_config.rs

```diff
@@ -0,0 +1,426 @@
+use core::fmt;
+use std::{
+    cmp,
+    collections::HashMap,
+    env, fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use serde::Deserialize;
+
+use crate::{
+    cache::{Interned, INTERNER},
+    config::{self, set},
+    flags::Flags,
+    t,
+    util::output,
+};
+
+/// The bare minimum config, suitable for `bootstrap-shim`, but sharing code with the main `bootstrap` binary.
```

Comment:
I think we should embed rationale here (i.e. in this comment or in a doc comment in this file) for:

* Why we have minimal vs. non-minimal config, if it all lives in bootstrap anyway.
* What the process for reviewers/authors should be when deciding where to put something.
  * What, if any, are implications of that choice? How permanent is it, in terms of breaking users if we move from one to the other?

(This will help me review the code as well).